contribute/testing.md

```diff
@@ -0,0 +1,230 @@
+# Beam Testing Documentation
+
+# Overview
```

Comment:
Struggling a little bit with the order of the file.

Don't understand the connection between the matrix and the testing types and the precommit/post commit. For a given row -- do I want/need to know which testing type it is or when it runs?